kanidmd/src/lib/idm/server.rs

```diff
@@ -898,14 +903,8 @@ impl<'a> IdmServerProxyWriteTransaction<'a> {
         // check a password badlist to eliminate more content
         // we check the password as "lower case" to help eliminate possibilities
         let lc_password = PartialValue::new_iutf8(cleartext);
-        let badlist_entry = self
-            .qs_write
-            .internal_search_uuid(au, &UUID_SYSTEM_CONFIG)
-            .map_err(|e| {
-                ladmin_error!(au, "Failed to retrieve system configuration {:?}", e);
-                e
-            })?;
-        if badlist_entry.attribute_value_pres("badlist_password", &lc_password) {
+        let pw_badlist_cache = &*self.pw_badlist_cache.read();
```

Comment:
If you change the type to a HashSet<string> you can do the check pw_badlist_cache.contains(cleartext) before the conversion to lc_password :)